codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ClientUpgradeCodec.java

```diff
@@ -74,37 +87,63 @@ public Http2ClientUpgradeCodec(Http2ConnectionHandler connectionHandler) {
      * @param connectionHandler the HTTP/2 connection handler
      */
     public Http2ClientUpgradeCodec(String handlerName, Http2ConnectionHandler connectionHandler) {
-        this(handlerName, connectionHandler, connectionHandler);
+        this(handlerName, connectionHandler, connectionHandler, null);
+    }
+
+    /**
+     * Creates the codec providing an upgrade to the given handler for HTTP/2.
+     *
+     * @param handlerName the name of the HTTP/2 connection handler to be used in the pipeline,
+     *                    or {@code null} to auto-generate the name
+     * @param connectionHandler the HTTP/2 connection handler
+     */
+    public Http2ClientUpgradeCodec(String handlerName, Http2ConnectionHandler connectionHandler,
+        Http2MultiplexHandler http2MultiplexHandler) {
```

Comment:
I wonder if we should make this more consistent with what we have on the server side:

https://github.com/netty/netty/blob/4.1/codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ServerUpgradeCodec.java#L107

Basically:

public Http2ClientUpgradeCodec(Http2FrameCodec http2Codec, ChannelHandler... handlers) {
     ...
}